config/initializers/devise.rb

```diff
@@ -270,4 +283,18 @@ def to_mobile
   # When using omniauth, Devise cannot automatically set Omniauth path,
   # so you need to do it manually. For the users scope, it would be:
   # config.omniauth_path_prefix = '/my_engine/users/auth'
+
+  # implement pam support analog to mastodon
+  if AppConfig.pam.enable
+    config.pam_authentication     = true
+    config.usernamefield          = 'username'
+    config.emailfield             = 'email'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.